packages/angular/src/migrations/update-13-3-0/update-mfe-webpack-config.ts

```diff
@@ -0,0 +1,158 @@
+import type { Tree } from '@nrwl/devkit';
+import {
+  logger,
+  formatFiles,
+  updateProjectConfiguration,
+  readProjectConfiguration,
+} from '@nrwl/devkit';
+import { tsquery } from '@phenomnomnominal/tsquery';
+import { forEachExecutorOptions } from '@nrwl/workspace/src/utilities/executor-options-utils';
+import { Identifier, Node, StringLiteral } from 'typescript';
+
+export default async function (tree: Tree) {
+  const NRWL_WEBPACK_BROWSER_BUILDER = '@nrwl/angular:webpack-browser';
+  const CUSTOM_WEBPACK_OPTION = 'customWebpackConfig';
+
+  const projects = [];
+  forEachExecutorOptions(
+    tree,
+    NRWL_WEBPACK_BROWSER_BUILDER,
+    (opts, projectName) => {
+      // Update the webpack config
+      const webpackPath = opts[CUSTOM_WEBPACK_OPTION]?.path;
+      if (!tree.exists(webpackPath)) {
+        logger.warn(
+          `Webpack config file for project: ${projectName} does not exist. Skipping project.`
+        );
+        return;
+      }
+      const webpackConfig = tree.read(webpackPath, 'utf-8');
+      if (!webpackConfig.includes('ModuleFederationPlugin')) {
+        return;
+      }
+
+      projects.push(projectName);
+
+      let updatedWebpackFile = addExperimentsObject(webpackConfig);
+      updatedWebpackFile = addLibraryModuleType(updatedWebpackFile);
+      updatedWebpackFile = removeRemoteName(updatedWebpackFile);
+      tree.write(webpackPath, updatedWebpackFile);
+    }
+  );
+  addPublicHost(tree, projects);
+
+  await formatFiles(tree);
+}
+
+export function addExperimentsObject(webpackConfig: string) {
+  const WEBPACK_EXPERIMENTS_QUERY =
+    'PropertyAssignment > Identifier[name=experiments] ~ ObjectLiteralExpression';
+  const ast = tsquery.ast(webpackConfig);
+  const webpackExperimentsNode = tsquery(ast, WEBPACK_EXPERIMENTS_QUERY);
+
+  if (webpackExperimentsNode && webpackExperimentsNode.length > 0) {
+    if (webpackExperimentsNode[0].getText().includes('outputModule')) {
+      return webpackConfig;
+    }
+  }
+
+  const WEBPACK_EXPORT = 'module.exports = {';
+  const openingBracePos =
+    webpackConfig.indexOf(WEBPACK_EXPORT) + WEBPACK_EXPORT.length;
+
+  return `${webpackConfig.slice(0, openingBracePos)}
+  experiments: {
+    outputModule: true  
+  },${webpackConfig.slice(openingBracePos)}`;
```

Comment:
This could result in the `experiments` property being duplicated. If someone has:

```javascript
// webpack.config.js

module.exports = {
  ...
  experiments: {
    // other properties here, no outputModule
  },
  ...
};
```

Because we are always inserting an `experiments` object at the start, it would result in:

```javascript
// webpack.config.js

module.exports = {
  experiments: {
    outputModule: true
  },
  ...
  experiments: {
    // other properties here, no outputModule
  },
  ...
};
```